imap/caldav_db.c

```diff
@@ -1179,20 +1179,18 @@ EXPORTED int caldav_writeical(struct caldav_db *caldavdb, struct caldav_data *cd
     }
     cdata->comp_flags.transp = transp;
 
-    /* Determine privacy */
-    cdata->comp_flags.privacy = CAL_PRIVACY_PUBLIC;
-    prop = icalcomponent_get_first_property(comp, ICAL_CLASS_PROPERTY);
-    if (prop) {
-        switch (icalproperty_get_class(prop)) {
-            case ICAL_CLASS_CONFIDENTIAL:
-                cdata->comp_flags.privacy = CAL_PRIVACY_SECRET;
-                break;
-            case ICAL_CLASS_PRIVATE:
-            case ICAL_CLASS_X:
-                cdata->comp_flags.privacy = CAL_PRIVACY_PRIVATE;
-                break;
-            default:
-                ; // ignore
+    /* Determine JSCalendar privacy */
+    for (prop = icalcomponent_get_first_property(comp, ICAL_X_PROPERTY);
```

Comment:
Could probably use icalcomponent_get_x_property_by_name() in ical_support.c for this